src/nodes/display/BloomNode.js

```diff
@@ -307,7 +307,7 @@ class BloomNode extends TempNode {
 
 			} );
 
-			return vec4( diffuseSum.div( weightSum ), 1.0 );
+			return vec4( diffuseSum.div( weightSum ), 0.0 ); // alpha zero to honor transparent background
```

Comment:
Transparent bloom was discussed extensively in #14104. We did not add an official solution since the bloom pass can't honor the luminance from the HTML background which leads to wrong result.

In any event, I think just using `0` isn't right. You might want to try one of the solutions in further down the issue e.g. https://github.com/mrdoob/three.js/issues/14104#issuecomment-1536582334.

Reply:
>the bloom pass can't honor the luminance from the HTML background

Can you please explain what you mean by "honor" in this case? What are the artifacts you do not like?

Note, this is the `BloomNode` implementation I am referring to -- not the `BloomPass` implementation.

Reply:
> Can you please explain what you mean by "honor" in this case? What are the artifacts you do not like?

`BloomNode` can't extract the luminance from a HTML background. But it can extract the luminance of a transparent canvas.

To the viewer it does not matter whether parts of the colors come from the HTML background or the canvas. So bloom is expected to work on the entire image, not just parts of it.

Reply:
In any event, I think it's okay to support for alpha. Hopefully, users know what they are doing. I think `seperableBlurPass` should look like so though:
```js
const seperableBlurPass = Fn( () => {

	const weightSum = gaussianCoefficients.element( 0 ).toVar();
	const diffuseSum = sampleTexel( uvNode ).mul( weightSum ).toVar();

	Loop( { start: int( 1 ), end: int( kernelRadius ), type: 'int', condition: '<' }, ( { i } ) => {

		const x = float( i );
		const w = gaussianCoefficients.element( i );
		const uvOffset = direction.mul( invSize ).mul( x );
		const sample1 = sampleTexel( uvNode.add( uvOffset ) );
		const sample2 = sampleTexel( uvNode.sub( uvOffset ) );
		diffuseSum.addAssign( add( sample1, sample2 ).mul( w ) );
		weightSum.addAssign( float( 2.0 ).mul( w ) );

	} );

	return vec4( diffuseSum.div( weightSum ) );

} );

```

Reply:
>BloomNode can't extract the luminance from a HTML background

If you want bloom to apply to the background, add the texture to the _scene background_ -- instead of the canvas background.

I would not view this as a limitation.

>In any event, I think it's okay to support for alpha.

Good. I think it looks great.

Reply:
As mentioned by @mrdoob, alpha should be blurred, too, which does not happen with this PR.

Reply:
Converting this PR to a draft PR. I think this needs more thought and testing before a merge can happen.

Reply:
Actually, it seems this change breaks `webgpu_postprocessing_bloom`.

Reply:
Converting this PR to a draft PR. I think this needs more thought and testing before a merge can happen.

Do you mind sharing a fiddle that shows how the current implementation of `BloomNode` fails with a transparent canvas and colored HTML background? I want to try the solution of https://github.com/mrdoob/three.js/issues/14104#issuecomment-1536582334 which originally does not require a change to `UnrealBloomPass`.

Reply:
@Mugen87 You are correct. Somehow my testing went awry... somewhere. So I am going to stop working on this and move on to something else.

>The colors of a HTML background can be produced in various ways and there is no way to always replicate the results with Scene.background.

What would be an example of a canvas texture that cannot be converted to a texture and used as a scene background texture?

Reply:
The background could be any HTML content if you style the renderer's canvas accordingly. E.g.

https://jsfiddle.net/hjerL1d7/

Reply:
I see what you mean, now.